### PR TITLE
Update dataset definitions for the measurePhotonTransferCurveTask

### PR DESCRIPTION
**Pull request by mfisherlevine**

### CODE REVIEW

File: policy/datasets.yaml [hunk 1]
```diff
@@ -1398,17 +1398,10 @@ measurePhotonTransferCurve_metadata:
...
-    description: "Persisted output data (gain and noise) from the MeasurePhotonTransferCurveTask"
-    persistable: ignored
-    python: lsst.cp.pipe.ptc.PtcDataset
-    storage: PickleStorage
-    tables: 'raw'
-    template: ''
-measurePhotonTransferCurveDatasetAll:
+photonTransferCurveDataset:
     description: "Persisted output data from the MeasurePhotonTransferCurveTask"
     persistable: ignored
```
Comment (plazas) [inline policy/datasets.yaml]: What does this "persistable" option mean and why is it set to "ignored"? The description says that you are persisting data.

File: policy/datasets.yaml [hunk 2]
```diff
@@ -1398,17 +1398,10 @@ measurePhotonTransferCurve_metadata:
     storage: YamlStorage
     python: lsst.daf.base.PropertySet
     template: ''
-measurePhotonTransferCurveGainAndNoise:
-    description: "Persisted output data (gain and noise) from the MeasurePhotonTransferCurveTask"
-    persistable: ignored
-    python: lsst.cp.pipe.ptc.PtcDataset
-    storage: PickleStorage
-    tables: 'raw'
```
Comment (plazas) [inline policy/datasets.yaml]: I noticed that you left out "tables: raw" in the new definition. However, Robert had to add this again to fix a runtime error (see DM-22227).

File: policy/datasets.yaml (hunk 1 above)
Comment (mfisherlevine) [reply policy/datasets.yaml]: See OOB communication "explaining" this (in the loosest possible sense of "explain").

File: policy/datasets.yaml (hunk 2 above)
Comment (mfisherlevine) [reply policy/datasets.yaml]: This is a good point, I'll check this out.

Comment (mfisherlevine) [reply policy/datasets.yaml]: Excellent catch, this totally needed fixing to not revert those changes, thank you!

**Pull request merged**